virtcontainers/qemu.go

```diff
@@ -1553,6 +1567,23 @@ func (q *qemu) resizeVCPUs(reqVCPUs uint32) (currentVCPUs uint32, newVCPUs uint3
 	return currentVCPUs, newVCPUs, nil
 }
 
+func (q *qemu) updateMemoryBalloon(sizeMB uint32) error {
+
+	if q.state.hasVFIODevice {
+		// Reference:
+		// https://www.ibm.com/support/knowledgecenter/en/SSZJY4_3.1.0/liabp/liabpmemoryballooning.htm
+		hostMB, err := q.hostMemMB()
```

Comment:
@bergwolf the idea is to set the balloon to very large size, this is to avoid memory is returned back to the host on VFIO case, this is patched in qemu +`3.1.0` , I added comments in the code. 
\cc @awilliam


@bergwolf  For `guest maximum dimm` I need to do a qmp call that will give me inforamation in bytes, while we dont move all the code logic to use only bytes I dont want to add more code to convert from MB to Bytes and and vice versa. 

Also, I added comments about why I dont add a `currentMemory == reqMemMB` validation.

Reply:
It looks a bit strange to me as well, it's relying on an assigned device VM not supporting memory over-commit and thus never having a footprint larger than host memory.  But an mdev VM doesn't necessarily pin all of memory and therefore could be over-committed to some extent.  What happens if you have such a VM, with a fully deflated balloon, and then call this?  Seems it could actually cause a balloon inflation.  The QEMU balloon inhibitor takes the approach that we do nothing when an inhibitor is added.  Anything that gets pinned by vfio will be re-allocated in the VM address space by the act of that pinning, it's only a matter that the guest balloon driver is possibly still holding onto pages, keeping the balloon inflated, which are now backed in the VM address space anyway.  This also maintains the appearance of ballooning within the guest, guest available memory changes with the balloon, only the memory isn't freed back to the host.  The important part is that the inhibitor prevents further inflation of the balloon, which can zap pages from the VM which are already pinned by vfio.  If you want to make sure the guest balloon driver is releasing all pages when a vfio device is added, perhaps figure out the data type used for the QMP interface and pass the maximum value it supports rather than creating this dependency on host memory size.

Reply:
(yikes, 2 years later... )

In the context of Kata, I don't think we'll be in a position of memory over-commit for a pod/container. If there aren't any other side effects, I'm not sure we should be _that_ concerned, other than it looking a bit odd!